### PR TITLE
Update the tutorials

### PR DESCRIPTION
**Pull request by blink1073**
cf #586.

### CODE REVIEW

File: tutorial/patterns.md [hunk 1]
```diff
@@ -91,3 +91,11 @@ the model cannot be provided in the constructor is the initialization
 required for a model may have to call methods that are subclassed.
 The subclassed methods would be called before the subclass constructor has
 finished evaluating, resulting in undefined state.
+
+
+## Data Structures
+
+Prefer to use Phosphor Vectors over JavaScript Arrays for internal use,
```
Comment (afshin) [inline tutorial/patterns.md]:
```markdown
[Phosphor `Vector`](http://phosphorjs.github.io/phosphor/api/classes/_collections_vector_.vector.html) over JavaScript `Array`
```

File: tutorial/patterns.md [hunk 2]
```diff
@@ -91,3 +91,11 @@ the model cannot be provided in the constructor is the initialization
...
+and expose Vectors as ISequences for external use.  This allows us to use
+the Phosphor algorithms which provide advanced functionality over what is
+offered by the builtin Arrays.
```
Comment (afshin) [inline tutorial/patterns.md]: I'd say "native" or "built-in". Sorry for being nit picky re: verbiage. Feel free to disregard.

Comment (blink1073) [inline tutorial/patterns.md]: Pick away my friend.

**Comment by afshin** [comment]
👍

**Pull request merged**